File: examples/eAPI v1.8/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/OneclickEchoReq.cs
using System.Text;
using Newtonsoft.Json;
namespace CsobGatewayClientExample.Communication.DataObjects
{
    public class OneclickEchoReq : SignBaseRequest
    {
        [JsonProperty("origPayId")]
        public string OrigPayId { get; set; }

        public OneclickEchoReq()
        {

        }

        public OneclickEchoReq(string merchantId, string OrigPayId)
        {
            this.MerchantId = merchantId;
            this.OrigPayId = OrigPayId;
        }
        
        public override string ToSign()
        {
            var sb = new StringBuilder();
            Add(sb, MerchantId);
            Add(sb, OrigPayId);
            Add(sb, DateTime);
            DeleteLast(sb);
            return sb.ToString();
        }
    }
}

File: examples/eAPI v1.8/java/eAPI-v18-connector/src/main/java/cz/monetplus/mips/eapi/v18/connector/entity/OneclickEchoReq.java
package cz.monetplus.mips.eapi.v18.connector.entity;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class OneclickEchoReq extends SignBase {

	private static final long serialVersionUID = -438284924457338635L;

	@XmlElement(nillable=false)
	public String merchantId;
	
	@XmlElement(nillable=false)
	public String origPayId;
	
	public OneclickEchoReq() {
		
	}

	public OneclickEchoReq(String merchantId, String origPayId) {
		this.merchantId = merchantId;
		this.origPayId = origPayId;
	}

	@Override
	public String toSign() {
		StringBuilder sb = new StringBuilder(); 
		add(sb, merchantId);
		add(sb, origPayId);
		add(sb, dttm);
		deleteLast(sb);
		return sb.toString();
	}

}
